pkg/utils/utils.go

```diff
@@ -257,3 +268,121 @@ func DeleteDoubleWhiteSpace(slice []string) []string {
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Service":
+		object, err := KubeClient.CoreV1().Services(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Secret":
+		object, err := KubeClient.CoreV1().Secrets(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "ConfigMap":
+		object, err := KubeClient.CoreV1().ConfigMaps(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "DaemonSet":
+		object, err := KubeClient.ExtensionsV1beta1().DaemonSets(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Ingress":
+		object, err := KubeClient.ExtensionsV1beta1().Ingresses(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+
+	case "ReplicaSet":
+		object, err := KubeClient.ExtensionsV1beta1().ReplicaSets(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Deployment":
+		object, err := KubeClient.ExtensionsV1beta1().Deployments(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Job":
+		object, err := KubeClient.BatchV1().Jobs(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "Role":
+		object, err := KubeClient.RbacV1().Roles(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "RoleBinding":
+		object, err := KubeClient.RbacV1().RoleBindings(obj.InvolvedObject.Namespace).Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "ClusterRole":
+		object, err := KubeClient.RbacV1().ClusterRoles().Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	case "ClusterRoleBinding":
+		object, err := KubeClient.RbacV1().ClusterRoleBindings().Get(obj.InvolvedObject.Name, metaV1.GetOptions{})
+		if err == nil {
+			return object.ObjectMeta.Annotations
+		}
+		log.Logger.Error(err)
+	}
+
+	return map[string]string{}
```

Comment:
This adds up with the discussion regarding the unit test case for object deleted